todo-backend/README.adoc

```diff
@@ -0,0 +1,452 @@
+
+* You must be logged in OpenShift and have an `oc` client to connect to OpenShift
+* https://helm.sh[Helm] must be installed to deploy the backend on OpenShift.
+
+Once you have installed Helm, you need to add the repository that provides Helm Charts for {productName}:
+
+ifndef::ProductRelease[]
+[source,options="nowrap"]
+----
+$ helm repo add wildfly https://docs.wildfly.org/wildfly-charts/
+"wildfly" has been added to your repositories
+$ helm search repo wildfly
+NAME                    CHART VERSION   APP VERSION     DESCRIPTION
+wildfly/wildfly         1.1.0           22.0            Build and Deploy WildFly applications on OpenShift
+----
+endif::[]
+ifdef::ProductRelease[]
+[source,options="nowrap"]
+----
+$ helm repo add jboss-eap https://jbossas.github.io/eap-charts/
+"jboss-eap" has been added to your repositories
+$ helm search repo jboss-eap
+NAME                    CHART VERSION   APP VERSION     DESCRIPTION
+jboss-eap/eap7          1.0.0           7.4             A Helm chart to build and deploy EAP 7 applic...
+----
+endif::[]
+
+
+=== Deploy a PostgreSQL Database on OpenShift
+
+[source,options="nowrap"]
+----
+$ oc new-app postgresql-ephemeral \
+   -p DATABASE_SERVICE_NAME=todos-db \
+   -p POSTGRESQL_DATABASE=todos
+----
+
+This will create a PostgreSQL database named `todos` on OpenShift that can be accessed on the port `5432` on the service `todos-db`.
+We don't need to copy the credentials to connect to the database as we will retrieve them later using the `todos-db` secret that was created
+when the database is deployed.
+
+ifndef::ProductRelease[]
+=== Build and Deploy the Backend on OpenShift with Bootable Jar
+
+The backend will be built and deployed on OpenShift with a Helm Chart for {productName}.
+
+[source,options="nowrap"]
+----
+$ helm install todo-backend -f todo-backend-bootable-jar.yaml wildfly/wildfly
+NAME: todo-backend
+...
+STATUS: deployed
+REVISION: 1
+----
+
+The Helm Chart for this quickstart contains all the information to build an image from the source code using Bootable Jar:
+
+[source,options="nowrap"]
+----
+build:
+  uri: https://github.com/wildfly/quickstart.git
+  mode: bootable-jar
+----
+endif::[]
+
+=== Build and Deploy the Backend on OpenShift with {productName} S2I
+
+The backend will be built and deployed on OpenShift with a Helm Chart for {productName}.
+
+ifndef::ProductRelease[]
+[source,options="nowrap"]
+----
+$ helm install todo-backend -f todo-backend-s2i.yaml wildfly/wildfly
+NAME: todo-backend
+...
+STATUS: deployed
+REVISION: 1
+----
+endif::[]
+ifdef::ProductRelease[]
+[source,options="nowrap"]
+----
+$ helm install todo-backend -f todo-backend-s2i.yaml jboss-eap/eap7
+NAME: todo-backend
+...
+STATUS: deployed
+REVISION: 1
+----
+endif::[]
+
+The Helm Chart for this quickstart contains all the information to build an image from the source code using S2I:
+
+ifndef::ProductRelease[]
+[source,options="nowrap"]
+----
+build:
+  uri: https://github.com/wildfly/quickstart.git
```

Comment:
@emmartins @jmesnil not sure if we want to have it hardcoded here, there is for sure a property for this which would make these  to `ifndef::ProductRelease[]` + `ifdef::ProductRelease[]` redundant

Note: it will still be hardcoded in the `todo-backend-s2i.yaml` though.

Reply:
the content of the YAML file will be different in EAP as it will require an additional env var to provision the datasource feature pack.